src/allotropy/allotrope/schema_parser/generate_schemas.py

```diff
@@ -82,9 +82,11 @@ def generate_schemas(root_dir: Path) -> int:
         print(f"Generating models for schema: {rel_schema_path}...")  # noqa: T201
         schema_name = rel_schema_path.stem
         schema_path = os.path.join(root_dir, SCHEMA_DIR_PATH, rel_schema_path)
-        output_file = re.sub(
-            "/|-", "_", f"{rel_schema_path.parent}_{schema_name}.py"
-        ).lower()
+        output_file = (
+            re.sub("/|-", "_", f"{rel_schema_path.parent}_{schema_name}.py")
+            .replace(".embed.", "_embed_")
```

Comment:
I trust this was needed, but where/why?

Reply:
The model generation fails at the linting method with `.embed.` in the name, derived from `.embed.` in the json schema name. Ruff complained about an invalid name.

Reply:
This isn't specified in the current contributing / readme (as we don't expect outside contributors to be adding schemas just yet), but we just don't include the `embed` in the name (it's the same schema, just without external links)

Reply:
@nathan-stender Confirming that I should rename `light-obscuration.embed.schema.json` to `light-obscuration.schema.json`

Reply:
Yup!